src/runtime/store/index.ts

```diff
@@ -81,24 +99,26 @@ export function writable<T>(value?: T, start: StartStopNotifier<T> = noop): Writ
 					subscriber_queue.length = 0;
 				}
 			}
+		} else {
+			revalidate_writable();
```

Comment:
I suspect that this logic might be incorrect; I'll try to come up with proof (i.e., a use case where this ends up donig the wrong thing). Once I come up with an example, I'll put it in a REPL and link it here.